lib/Models/MapboxVectorTileCatalogItem.ts

```diff
@@ -0,0 +1,149 @@
+import { computed, runInAction, observable, action, autorun } from "mobx";
+import AsyncMappableMixin from "../ModelMixins/AsyncMappableMixin";
+import CatalogMemberMixin from "../ModelMixins/CatalogMemberMixin";
+import UrlMixin from "../ModelMixins/UrlMixin";
+import MapboxVectorTileCatalogItemTraits from "../Traits/MapboxVectorTileCatalogItemTraits";
+import CreateModel from "./CreateModel";
+
+import Rectangle from "terriajs-cesium/Source/Core/Rectangle";
+import LoadableStratum from "./LoadableStratum";
+import { BaseModel } from "./Model";
+import StratumOrder from "./StratumOrder";
+import ImageryLayerFeatureInfo from "terriajs-cesium/Source/Scene/ImageryLayerFeatureInfo";
+import ImageryProvider from "terriajs-cesium/Source/Scene/ImageryProvider";
+import MapboxVectorTileImageryProvider from "../Map/MapboxVectorTileImageryProvider";
+import i18next from "i18next";
+import isDefined from "../Core/isDefined";
+import { VectorTileFeature } from "@mapbox/vector-tile";
+import { clone } from "lodash-es";
+
+class MapboxVectorTileLoadableStratum extends LoadableStratum(
+  MapboxVectorTileCatalogItemTraits
+) {
+  static stratumName = "MapboxVectorTileLoadable";
+
+  constructor(readonly item: MapboxVectorTileCatalogItem) {
+    super();
+  }
+
+  duplicateLoadableStratum(newModel: BaseModel): this {
+    return new MapboxVectorTileLoadableStratum(
+      newModel as MapboxVectorTileCatalogItem
+    ) as this;
+  }
+
+  static async load(item: MapboxVectorTileCatalogItem) {
+    return new MapboxVectorTileLoadableStratum(item);
+  }
+
+  // @computed get legends() {
```

Comment:
As stated in decription:
I'm not sure how to handle `legends` set by manual properties like `lineColor` and `fillColor`

Reply:
I think a computed on MapboxVectorTileCatalogItem is best. Just check `super.legends` and return that if it exists. See e.g. https://github.com/TerriaJS/terriajs/blob/b131abaa77fa8aa9a6f89238951f33cbf0ac4130/lib/Models/ArcGisMapServerCatalogItem.ts#L435-L446

Reply:
I have put this inside the `LoadableStratum`. I think extending like above is easy for `primitiveTrait`s as they return a simple type `T`, but it is difficult for more complex traits returning a `Model<T>`.